src/viam/sdk/components/board/board.hpp

```diff
@@ -0,0 +1,162 @@
+/// @file components/board/board.hpp
+///
+/// @brief Defines a `Board` component.
+#pragma once
+
+#include <string>
+
+#include <viam/api/component/board/v1/board.pb.h>
+
+#include <viam/sdk/common/proto_type.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource_manager.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup Board Classes related to the `Board` component.
+
+/// @class BoardSubtype
+/// @brief Defines a `ResourceSubtype` for the `Board` component.
+/// @ingroup Board
+class BoardSubtype : public ResourceSubtype {
+   public:
+    explicit BoardSubtype(const google::protobuf::ServiceDescriptor* service_descriptor);
+    std::shared_ptr<ResourceServer> create_resource_server(
+        std::shared_ptr<ResourceManager> manager) override;
+    std::shared_ptr<Resource> create_rpc_client(std::string name,
+                                                std::shared_ptr<grpc::Channel> chan) override;
+};
+
+/// @class Board board.hpp "components/board/board.hpp"
+/// @brief Represents a physical board with gpio pins, digital interrupts, and analog voltage
+/// reading
+/// @ingroup Board
+///
+/// This acts as an abstract base class to be inherited from by any drivers representing
+/// specific board implementations. This class cannot be used on its own.
+class Board : public Component {
+   public:
+    /// @brief Represents the value received by the registered analog to digital converter (ADC).
+    /// The range and conversion mechanism to voltage will vary depending on the specific ADC
+    /// registered to the pin. Consult your ADC's documentation and Viam's `Board` documentation for
+    /// more details.
+    typedef int32_t analog_value;
```

Comment:
Name justification: Very little. This is what it is called in the other sdks but that doesn't mean that it isn't confusing

Typedef justification: Treating this as a special type may give us a bit more flexibility going forward (on major version bumps), and may prevent surprising rounding errors for customers.

Reply:
Switched to `using` and resolving.